torch/_decomp/__init__.py

```diff
@@ -385,6 +385,7 @@ def core_aten_decompositions() -> Dict[torch._ops.OperatorBase, Callable]:
             aten.renorm,
             aten.renorm_,
             aten.replication_pad2d,
+            aten.roll,
```

Comment:
Should this be inductor-only to not incur the revert from export.

Reply:
That's the thing. I think this should be a core decomposition. As the lowering implies, this was not a core decomp from the get go just because we were missing the constant propagation pass that you implemented.

Reply:
I'm not comfortable approving without resolving this.

Reply:
I'm not comfortable approving without resolving this.